wgpu-hal/src/dx12/mod.rs

```diff
@@ -896,6 +898,7 @@ struct PipelineLayoutShared {
     signature: Option<Direct3D12::ID3D12RootSignature>,
     total_root_elements: RootIndex,
     special_constants_root_index: Option<RootIndex>,
+    special_constants_cmd_signatures: Option<CommandSignatures>,
```

Comment:
suggestion: It seems that `special_constants_*` members of this field are all or nothing, contrary to the individual `Option`ality for each. I think it'd be worth merging these behind a single `Option`, probably as a `struct`.

I have a `fixup!` ready to go for this.

Reply:
Would you like the change to get pushed in here, then? If you have concrete plans to still remove it, then 🤷🏻‍♂️ whatevs.

Reply:
Pushed, will let you reject if you don't want it: 00b542f8